### ISSUE TITLE
BUG: Certain outbound rules stopped working in v4.0.1, the problem narrowed down to nekobox.exe module (Windows 11)

### ISSUE THREAD

**Issue by Yuri4264** [MatsuriDayo/nekoray#1540]
## Describe the problem
I am using a routing rule 
          {   "rules": [    {   "domain_suffix": [     ".ru"    ],   "outbound": "direct"     }   ]   }

It worked nicely up to and including v4.0-beta4. 

Expected behavior:  The domains ending with suffix .ru should be routed directly, bypassing VPN protocol.

Actual behavior: V4.0.1 sends the packets via the VPN path, that is, the traffic for sites like www.rbc.ru still goes through the VPN stream.

## How to reproduce  
To reproduce the bug you have to be in the .RU zone, of course, and use the rule shown above. I managed to narrow down the culprit to nekobox.exe module: 
if you install V4.0.1 and then replace nekobox.exe with the previous version from the package v4.0-beta4 (nekobox.exe sized  3,315,712 bytes with timestamp 10/09/24  04:41am), everything works as expected re this rule.

Provide helpful screenshots, videos, text descriptions, subscription links, etc.

## log

If you have logs, please upload them. Please see the detailed steps for exporting logs in the documentation.

**Comment by devopg** [comment]
```
                "domain_suffix": [
                    "ru"
                ],
```

work fine
maybe problem with dot !?

**Comment by Yuri4264** [comment]
Nope, this does not help either. I tried to remove dot, tested modified rule with nekobox.exe v4.0-b4 - works just fine, no difference, so v4.0-b4 does not care about dot. 
Nekobox v4.0.1 does not work with both settings - with or without dot. The error I am getting while trying to connect to some *.ru  site is "Error code: PR_END_OF_FILE_ERROR".

**Comment by dtinside** [comment]
+1 

Old rules from 3.x version don't work in 4.0.1.


Example
```

        {
            "domain": [
                "regexp:.*\\.ru$"
            ],
            "outboundTag": "direct",
            "type": "field"
        },
```
Get Error on startup

**Comment by laniakeadev** [comment]
Same
